src/structures/Message.js

```diff
@@ -560,7 +560,7 @@ class Message extends Base {
     return Boolean(
       !this.deleted &&
         (this.author.id === this.client.user.id ||
-          this.channel.permissionsFor?.(this.client.user)?.has(Permissions.FLAGS.MANAGE_MESSAGES)),
+          this.channel?.permissionsFor?.(this.client.user)?.has(Permissions.FLAGS.MANAGE_MESSAGES)),
```

Comment:
```suggestion
          this.channel?.permissionsFor(this.client.user)?.has(Permissions.FLAGS.MANAGE_MESSAGES)),
```
consistency, so it matches the rest (or change the other ones to match this one)

Reply:
I did notice that optional chaining didn't make much sense but I decided to leave it alone since it's not quite within the scope of the PR is it
In reality that will never not be a function I think so we can leave it like that

Reply:
actually DMChannels don't have the `permissionsFor` method

Reply:
Ohhh right, should I add this to the other methods then?

Reply:
well the other getters already ensure its a guild channel so shouldn't be needed for them

Reply:
Marking this as resolved then